src/nvim/api/vim.c

```diff
@@ -848,6 +848,24 @@ void nvim_set_current_win(Window window, Error *err)
   }
 }
 
+/// Create new empty buffer
+///
+/// @param listed whether the buffer should be listed
+/// @param[out] err Error details, if any
+/// @return the buffer handle or 0 when error
+Buffer nvim_create_buf(Boolean listed, Error *err)
+  FUNC_API_SINCE(5)
+{
+  try_start();
+  Buffer buffer = buflist_add(NULL, listed ? BLN_LISTED : 0);
```

Comment:
include `BLN_NOOPT` to be explicit?

Also just noticed we have not merged `BLN_NEW`, maybe it's useful. It's used in Vim for `term_start`:

```c
        /* Create a new buffer without a window. Make it the current buffer for
         * a moment to be able to do the initialisations. */              
        buf = buflist_new((char_u *)"", NULL, (linenr_T)0,
                                                         BLN_NEW | BLN_LISTED);
```

Reply:
`BLN_NEW` could maybe avoid the option change. The patch looks simple enough to try.